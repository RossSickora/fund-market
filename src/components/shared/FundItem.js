import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Paper } from '@material-ui/core';

const useStyles = theme => ({
    fundItem: {
        marginTop: '15px',
        width: '100%',
        padding: '5px',
        maxWidth: '800px'
    },
  });

class FundItem extends Component {
    state = {elevation: 2}
    render() { 
        const {classes} = this.props;
        return (
            <Paper
                className={classes.fundItem}
                elevation = {this.state.elevation}
                onMouseOver = {() => this.setState( {elevation: 6} )}
                onMouseOut = {() => this.setState( {elevation: 2} )}
            >
                Fund Item
            </Paper>
        )
    }
}

export default withStyles(useStyles)(FundItem);