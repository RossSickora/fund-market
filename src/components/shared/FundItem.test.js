import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils'
import renderer from 'react-test-renderer';
import FundItem from './FundItem';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FundItem />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// it('changes the elevation to 6 on mouse entering the component', (done) => {
//     const component = renderer.create(<FundItem/>,);

//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();


//     tree.props.onMouseOver();
//     tree = component.toJSON();
//     //expect(tree.children.elevation).toEqual(6);
//     expect(tree).toMatchSnapshot();

//     tree.props.onMouseOut();
//     tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });