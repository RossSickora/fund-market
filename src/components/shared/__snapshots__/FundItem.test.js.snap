// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the elevation to 6 on mouse entering the component 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation2 FundItem-fundItem-29 MuiPaper-rounded"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  Fund Item
</div>
`;

exports[`changes the elevation to 6 on mouse entering the component 2`] = `
<div
  className="MuiPaper-root MuiPaper-elevation6 FundItem-fundItem-29 MuiPaper-rounded"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  Fund Item
</div>
`;

exports[`changes the elevation to 6 on mouse entering the component 3`] = `
<div
  className="MuiPaper-root MuiPaper-elevation2 FundItem-fundItem-29 MuiPaper-rounded"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  Fund Item
</div>
`;
